controllers/grid/settings/sections/form/PKPSectionForm.inc.php

```diff
@@ -87,15 +87,9 @@ public function _getAssignedSubEditorIds($sectionId, $contextId) {
 			'assignedToSection' => $sectionId,
 		));
 
-		if (empty($subEditors)) {
-			return array();
-		}
-
-		$subEditorIds = array_map(function($subEditor) {
+		return array_map(function($subEditor) {
 			return (int) $subEditor->getId();
-		}, $subEditors);
-
-		return $subEditorIds;
+		}, iterator_to_array($subEditors));
```

Comment:
In these cases, we can replace the `array_map` with a `foreach`. I have brought in the use of `array_*` functions as a carry-over from JS, where they are elegant and useful. 

But in practice they're not that legible in PHP and if we're moving to having our objects as `iterator`s instead of `array`s, it makes even less sense to use them.

Reply:
Yeah, I like these constructs too, but unfortunately there's no built-in equivalent for Iterators in PHP. Discussed here: https://stackoverflow.com/questions/6566129/php-lazy-array-mapping